packages/shared/lib/services/connection.service.ts

```diff
@@ -861,6 +861,10 @@ class ConnectionService {
 
         const copy = { ...connection };
 
+        if (!connection.credentials || 'encrypted_credentials' in connection.credentials) {
```

Comment:
is it not `!('encrypted_credentials' in connection.credentials)`?

Reply:
if credentials are empty or still encrypted, it's mostly to avoid regression. When the type refactor will lend it will be more safely typed to avoid this error.